extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/drive/DriveExporter.java

```diff
@@ -0,0 +1,204 @@
+      TokensAndUrlAuthData authData, Optional<ExportInformation> optionalExportInformation)
+      throws Exception {
+    Drive driveInterface = getDriveInterface((authData));
+    List driveListOperation = driveInterface.files().list();
+    // If the folder Id isn't specified then use root
+    String parentId = "root";
+    if (optionalExportInformation.isPresent()) {
+      ExportInformation exportInformation = optionalExportInformation.get();
+      if (exportInformation.getPaginationData() != null) {
+        StringPaginationToken paginationToken =
+            (StringPaginationToken) exportInformation.getPaginationData();
+        driveListOperation.setPageToken(paginationToken.getToken());
+      }
+
+      if (exportInformation.getContainerResource() != null) {
+        BlobbyStorageContainerResource parent =
+            (BlobbyStorageContainerResource) exportInformation.getContainerResource();
+        parentId = parent.getId();
+      }
+    }
+    driveListOperation
+        .setFields("files(id, name, modifiedTime, mimeType)")
+        .setQ(String.format(DRIVE_QUERY_FORMAT, parentId));
+
+    ArrayList<DigitalDocument> files = new ArrayList<>();
+    ArrayList<BlobbyStorageContainerResource> folders = new ArrayList<>();
+
+    FileList fileList = driveListOperation.execute();
+
+    for (File file : fileList.getFiles()) {
+      if (FOLDER_MIME_TYPE.equals(file.getMimeType())) {
+        folders.add(new BlobbyStorageContainerResource(
+            file.getName(),
+            file.getId(),
+            null,
+            null
+        ));
+      } else if (FUSION_TABLE_MIME_TYPE.equals(file.getMimeType())) {
+        logger.warn("Exporting of fusion tables is not yet supported: {}", file);
+      } else if (MAP_MIME_TYPE.equals(file.getMimeType())) {
+        logger.warn("Exporting of maps is not yet supported: {}", file);
+      } else {
+        try {
+          InputStream inputStream;
+          String newMimeType = file.getMimeType();
+          if (EXPORT_FORMATS.containsKey(file.getMimeType())) {
+            newMimeType = EXPORT_FORMATS.get(file.getMimeType());
+            inputStream = driveInterface.files().export(file.getId(), newMimeType)
+                .executeMedia().getContent();
+          } else {
+            inputStream =
+                driveInterface.files().get(file.getId()).setAlt("media").executeMedia()
+                    .getContent();
+          }
+          jobStore.create(jobId, file.getId(), inputStream);
+          files.add(new DigitalDocument(
+              file.getName(),
+              file.getModifiedTime().toStringRfc3339(),
+              newMimeType,
+              file.getMimeType(),
+              file.getId()));
+        } catch (Exception e) {
+          logger.error("Couldn't export {} with error", file, e);
+        }
+      }
+      logger.info("Exported {}", file);
+    }
+
+    ResultType resultType = fileList.getFiles().isEmpty()
```

Comment:
nit: how about a helper function for hasMore(fileList) or shouldContinue(fileList)?